rust/src/verify.rs

```diff
@@ -526,4 +582,186 @@ mod tests {
+        let spec = &inner_spec_with_empty_child();
+        let non_empty_spec = api::iavl_spec().inner_spec.unwrap();
+
+        let cases = vec![
+            EmptyBranchCase {
+                op: ics23::InnerOp {
+                    prefix: vec![1_u8],
+                    suffix: spec.empty_child.clone(),
+                    hash: spec.hash,
+                },
+                spec: spec,
+                is_true: true,
+                is_left: false,
+            },
+            EmptyBranchCase {
+                op: ics23::InnerOp {
+                    prefix: vec![1_u8]
+                        .into_iter()
+                        .chain(vec![0_u8; 32].into_iter())
+                        .collect(),
+                    suffix: spec.empty_child.clone(),
+                    hash: spec.hash,
+                },
+                spec: spec,
+                is_true: true,
+                is_left: false,
+            },
+            EmptyBranchCase {
+                op: ics23::InnerOp {
+                    prefix: vec![1_u8],
+                    suffix: vec![0_u8; 32],
+                    hash: spec.hash,
+                },
+                spec: spec,
+                is_true: false,
+                is_left: false,
+            },
+            EmptyBranchCase {
+                op: ics23::InnerOp {
+                    prefix: vec![1_u8],
+                    suffix: vec![],
+                    hash: spec.hash,
+                },
+                spec: spec,
+                is_true: false,
+                is_left: false,
+            },
+            EmptyBranchCase {
+                op: ics23::InnerOp {
+                    prefix: vec![1_u8],
+                    suffix: spec
+                        .empty_child
+                        .clone()
+                        .into_iter()
+                        .chain(b"xxxx".to_vec())
+                        .collect(),
+                    hash: spec.hash,
+                },
+                spec: spec,
+                is_true: false,
+                is_left: false,
+            },
+            EmptyBranchCase {
+                op: ics23::InnerOp {
+                    prefix: vec![1_u8],
+                    suffix: vec![],
+                    hash: non_empty_spec.hash,
+                },
+                spec: &non_empty_spec,
+                is_true: false,
+                is_left: false,
+            },
+            EmptyBranchCase {
+                op: ics23::InnerOp {
+                    prefix: vec![1_u8]
+                        .into_iter()
+                        .chain(spec.empty_child.iter().cloned())
+                        .collect(),
+                    suffix: vec![],
+                    hash: spec.hash,
+                },
+                spec: spec,
+                is_true: true,
+                is_left: true,
+            },
+            EmptyBranchCase {
+                op: ics23::InnerOp {
+                    prefix: vec![1_u8]
```

Comment:
Works, but I think `[&[1u8], &spec.empty_child].concat().to_vec()` will do the same and is a bit easier to read.